trestle/transforms/implementations/osco.py

```diff
@@ -87,28 +127,434 @@ def _ingest_yaml(self, blob: str) -> Results:
+        self.scanner_version = args['scanner_version']
+        self.idref = args['idref']
+        self.version = args['version']
+        self.time = args['time']
+        self.result = args['result']
+        self.severity = args['severity']
+        self.weight = args['weight']
+
+    @property
+    def inventory_key(self):
+        """Derive inventory key."""
+        if self.host_name is None:
+            # OpenScap 1.3.3
+            rval = self.target + ':' + self.target_type
+        else:
+            # OpenScap 1.3.5
+            rval = self.host_name + ':' + self.target_type
+        return rval
+
+
+class ComplianceOperatorReport():
+    """Represents one report of OSCO data."""
+
+    def __init__(self, osco_xml: str) -> None:
+        """Initialize given specified args."""
+        self.osco_xml = osco_xml
+
+    def _get_version(self, root: Element) -> str:
+        """Extract version from the XML."""
+        value = None
+        for key, val in root.attrib.items():
+            if key == 'version':
+                value = val
+                break
+        return value
+
+    def _get_id(self, root: Element) -> str:
+        """Extract id from the XML."""
+        value = None
+        for key, val in root.attrib.items():
+            if key == 'id':
+                value = val
+                break
+        return value
+
+    def _get_target(self, root: Element) -> str:
+        """Extract target from the XML."""
+        value = None
+        for lev1 in root:
+            tag = _remove_namespace(lev1.tag)
+            if tag == 'target':
+                value = root.find(lev1.tag).text
+                break
+        return value
+
+    def _get_target_type(self, root: Element) -> str:
+        """Extract target_type from the XML."""
+        value = None
+        benchmark_href = self._get_benchmark_href(root)
+        if benchmark_href is not None:
+            value = benchmark_href.split('-')[1]
+        return value
+
+    def _get_benchmark(self, root: Element, kw) -> str:
+        """Extract benchmark from the XML."""
+        value = None
+        for lev1 in root:
+            tag = _remove_namespace(lev1.tag)
+            if tag == 'benchmark':
+                value = lev1.get(kw)
+                break
+        return value
+
+    def _get_benchmark_href(self, root: Element) -> str:
+        """Extract benchmark.href from the XML."""
+        return self._get_benchmark(root, 'href')
+
+    def _get_benchmark_id(self, root: Element) -> str:
+        """Extract benchmark.id from the XML."""
+        return self._get_benchmark(root, 'id')
+
+    def _get_fact(self, lev1: Element, kw: str) -> str:
+        """Extract fact from the XML."""
+        value = None
+        for lev2 in lev1:
+            tag = _remove_namespace(lev2.tag)
+            if tag == 'fact':
+                name = lev2.get('name')
+                if name == kw:
+                    value = lev2.text
+                    break
```

Comment:
is name sure to be present?

Reply:
I think this is OK. If name is not found, None comes back. The `kw` should never be None else that's a code bug not a data error.